### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by GaoShuang98** [VicenteVivan/geo-clip#4]
Dear author,
I appreciate your work and would like to get the details of Geoclip's training. Can you publish the complete training code? Thank you.

Also, in the geoclip code, self. opt is not defined in init. How can this be changed?

        def _dequeue_and_enqueue(self, gps):
            """ Update GPS queue

        Args:
            gps (torch.Tensor): GPS tensor of shape (batch_size, 2)
        """
            opt = self.opt
            gps_batch_size = gps.shape[0]
            batch_size = opt.batch_size

            gps_ptr = int(self.gps_queue_ptr)
            assert self.queue_size % batch_size == 0

            # Replace the GPS from ptr to ptr+batch_size (dequeue and enqueue)
            self.gps_queue[:, gps_ptr:gps_ptr + gps_batch_size] = gps.t()
            gps_ptr = (gps_ptr + batch_size) % self.queue_size  # move pointer
            self.gps_queue_ptr[0] = gps_ptr

In the following code, weather the self.gps of self._dequeue_and_enqueue(self.gps) should be modified to self.gps_queue?

        def append_gps_queue_features(self, gps_features):
            """ Compute the GPS queue features and append them to the given GPS features."""
            # Get the GPS queue features
            location_queue = self.gps_queue.t().detach()
            gps_queue_features = self.location_encoder(location_queue)
            gps_queue_features = F.normalize(gps_queue_features, dim=1)

            # Concatenate Features (GPS Features & GPS Queue Features)
            gps_features = torch.cat([gps_features, gps_queue_features], dim=0)

            # Update GPS queue
            self._dequeue_and_enqueue(self.gps) 

            return gps_features

Looking forward to your response, thank you.

**Comment by a-parida12** [comment]
@GaoShuang98 @VicenteVivan do you have a solution for this?

**Comment by GaoShuang98** [comment]
> @GaoShuang98 @VicenteVivan do you have a solution for this?

Not yet😅
